CMakeLists.txt

```diff
@@ -1,52 +1,195 @@
-link_directories(${CMAKE_INSTALL_PREFIX}/lib)
-
-# Recursively look for CMakeLists.txt in subdirs.
-add_subdirectory("build/xinclude")
-add_subdirectory("libmetis")
-add_subdirectory("programs")
+cmake_minimum_required(VERSION 3.15)
+# CMake version compatibility
+# TODO: Remove when cmake 3.25 is commonly distributed
+if (POLICY CMP0140)
+	cmake_policy(SET CMP0140 NEW)
+endif ()
+
+#[==============================================================================================[
+#                                   Basic project definition                                   #
+]==============================================================================================]
+
+list(APPEND CMAKE_MESSAGE_CONTEXT METIS)
+project(METIS
+		VERSION 5.3.0
+		DESCRIPTION "Serial Graph Partitioning and Fill-reducing Matrix Ordering"
+		HOMEPAGE_URL https://github.com/KarypisLab/METIS
+		LANGUAGES C
+)
+
+# Back-porting to PROJECT_IS_TOP_LEVEL to older cmake
+# TODO: Remove when requiring cmake >= 3.21
+if (NOT DEFINED Spglib_IS_TOP_LEVEL)
+	if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+		set(PROJECT_IS_TOP_LEVEL ON)
+	else ()
+		set(PROJECT_IS_TOP_LEVEL OFF)
+	endif ()
+endif ()
+
+# Specify C standard
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS ON)
+
+if (NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release)
+endif ()
+
+#[==============================================================================================[
+#                                            Options                                            #
+]==============================================================================================]
+
+include(CMakeDependentOption)
+include(FeatureSummary)
+option(METIS_TESTS "METIS: Build unit tests" ${PROJECT_IS_TOP_LEVEL})
+option(METIS_SHARED_LIBS "METIS: Build as a shared library" ${PROJECT_IS_TOP_LEVEL})
+option(METIS_INSTALL "METIS: Install project" ${PROJECT_IS_TOP_LEVEL})
+option(METIS_OpenMP "METIS: Enable OpenMP support" OFF)
+add_feature_info(METIS_OpenMP METIS_OpenMP "OpenMP support")
+option(METIS_PROGRAMS "METIS: Build executable programs" OFF)
+add_feature_info(METIS_PROGRAMS METIS_PROGRAMS "Build executable programs")
+option(METIS_IDX64 "METIS: Build with 64 bit ints" OFF)
+add_feature_info(METIS_IDX64 METIS_IDX64 "Build with 64 bit ints")
+option(METIS_REAL64 "METIS: Build with 64 bit floats" OFF)
+add_feature_info(METIS_REAL64 METIS_REAL64 "Build with 64 bit floats")
+
+#[==============================================================================================[
+#                                     Project configuration                                     #
+]==============================================================================================]
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+# Add some helper functions
+include(METIS)
+
+# Include basic tools
+include(FetchContent)
+if (METIS_INSTALL)
+	include(CMakePackageConfigHelpers)
+	if (UNIX)
```

Comment:
This should be unconditional, as you use e.g. `CMAKE_INSTALL_BINDIR` elsewhere. It's not a problem to use GNUInstallDirs on Windows.